src/test/scala/nl/knaw/dans/easy/schemaExamples/SchemaValidationFixture.scala

```diff
@@ -15,19 +15,128 @@
 import javax.xml.validation.{ Schema, SchemaFactory }
+import org.scalatest.prop.{ TableDrivenPropertyChecks, TableFor1 }
 import org.scalatest.{ FlatSpec, Matchers }
 
-import scala.util.{ Failure, Try }
-import scala.xml.{ Elem, SAXParseException }
+import scala.util.{ Failure, Success, Try }
+import scala.xml._
 
-trait SchemaValidationFixture extends FlatSpec with Matchers {
+trait SchemaValidationFixture extends FlatSpec with Matchers with TableDrivenPropertyChecks {
 
+  val publicSchema: String
+  val localSchemaFile: String
+  val examples: TableFor1[String]
 
+  private val publicEasySchemaBase = "//easy.dans.knaw.nl/schemas"
+  protected val httpsEasySchemaBase = s"https:$publicEasySchemaBase"
+
+  private lazy val triedPublicSchema: Try[Schema] = Try {
+    SchemaFactory
+      .newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI)
+      .newSchema(Array(new StreamSource(publicSchema.toString)).toArray[Source])
+  }
+  private lazy val triedLocalSchema: Try[Schema] = Try {
+    // lazy for two reasons:
+    // - schemaFile is set by concrete test class
+    // - postpone loading until actually validating
+    val xsdInputStream = File(localSchemaFile).contentAsString.replaceAll(
+      s"""schemaLocation="http:$publicEasySchemaBase""",
+      s"""schemaLocation="file://$schemaDir"""
+    ).inputStream
+    SchemaFactory
+      .newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI)
+      .newSchema(Array(new StreamSource(xsdInputStream)).toArray[Source])
+  }
+
+  "examples" should "be schema valid with local copy of easy-schema" in {
+    forEvery(examples) { example =>
+      val xml = XML.loadFile((exampleDir / example).toString())
+      easyLocationsIn(xml).foreach(File(_).toJava should exist)
+      assume(schemaIsOnline(triedLocalSchema)) // inside the loop to not block other errors
+      validate(triedLocalSchema, toLocalSchemas(xml)) shouldBe a[Success[_]]
+    }
+  }
+
+  it should "be schema valid with (unqualified) public schema" in {
+    // sub-schemas may not have an unqualified public schema
+    assume(schemaIsOnline(triedPublicSchema)) // outside the loop as we are only validating
+    assume(lastLocalIsPublic)
+    forEvery(examples) { example =>
+      val xml = XML.loadFile((exampleDir / example).toString())
+      validate(triedPublicSchema, xml.toString) shouldBe a[Success[_]]
+    }
+  }
+
+  private def lastLocalIsPublic: Boolean = {
+    // to ignore tests while easy-schema and easy-schema-examples are under parallel development
+    // and the new schema's are not yet published
+    // reported with: ...lastLocalIsPublic was false
+    val newPublicSchema = localSchemaFile.replace(schemaDir.toString(), httpsEasySchemaBase)
+    Try(new URL(newPublicSchema).openStream()).fold(_ => false, _ => true)
+  }
+
+  private def validate(schema: Try[Schema], xmlString: String): Try[Unit] = {
+    schema.map(_.newValidator().validate(new StreamSource(xmlString.inputStream))) match {
```

Comment:
this pattern match sounds like:

```scala
schema.map(_.newValidator().validate(new StreamSource(xmlString.inputStream)))
  .doIfFailure {
    case e: SAXParseException => showErrorWithSourceContext(xmlString, e)
  }
```

it's just a side-effect, you're performing, right?

Only you need to add the following to the `pom.xml` for this:

```xml
<dependency>
    <groupId>nl.knaw.dans.lib</groupId>
    <artifactId>dans-scala-lib_2.12</artifactId>
    <scope>test</scope>
</dependency>
```